src/Npgsql/NpgsqlDataSource.cs

```diff
@@ -23,8 +26,17 @@ public abstract class NpgsqlDataSource : DbDataSource
     /// </summary>
     internal NpgsqlConnectionStringBuilder Settings { get; }
 
+    internal NpgsqlDataSourceConfiguration Configuration { get; }
     internal NpgsqlLoggingConfiguration LoggingConfiguration { get; }
 
+    readonly Func<NpgsqlConnectionStringBuilder, CancellationToken, ValueTask<string>>? _periodicPasswordProvider;
+    readonly TimeSpan _periodicPasswordSuccessRefreshInterval, _periodicPasswordFailureRefreshInterval;
+
+    readonly Timer? _passwordProviderTimer;
+    readonly CancellationTokenSource? _timerPasswordProviderCancellationTokenSource;
+    volatile Task? _passwordRefreshTask;
```

Comment:
Doesn't have to be volatile? (otherwise by that same thought _password should be as well)

Reply:
Yeah, leftover from a previous state. Removing.